numba/unicode.py

```diff
@@ -1388,6 +1389,22 @@ def impl(s):
     return impl
 
 
+# https://github.com/python/cpython/blob/1d4b6ba19466aba0eb91c4ba01ba509acf18c723/Objects/unicodeobject.c#L12188-L12213    # noqa: E501
+@overload_method(types.UnicodeType, 'isprintable')
+def unicode_isprintable(data):
+    """Implements UnicodeType.isprintable()"""
+
+    def impl(data):
+        for i in range(len(data)):
```

Comment:
The CPython impl has a shortcut for single characters, perhaps implement that?

Reply:
As I see the shortcut exactly repeats body of the main loop. Do you think we still need to add the following code to the implementation?
```
if length == 1:
    code_point = _get_code_point(data, 0)
    if _PyUnicode_IsPrintable(code_point):
        return True
    return False
```

Reply:
@stuartarchibald What do you think about above comment?

Reply:
See https://github.com/numba/numba/pull/4843#discussion_r346745393